x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -867,6 +867,35 @@ public void testMatchCommand() throws Exception {
         // TODO Keep adding tests for all unsupported commands
     }
 
+    public void testCoalesceWithMixedNumericTypes() {
```

Comment:
Here can you experiment a bit more with some more exotic field types? (`short`, `byte`, `scaled_float`....) I'd adventure even further and suggest `counter` data types as well.

Reply:
Just found some tests that **do** have counter fields in them, so it should be easy to test `coalesce` with `counter` data types as well. One example is `testAggregateOnCounter`.

Tested this quickly and `from test | where coalesce(k8s.pod.network.tx) IS NULL` works and doesn't throw any errors and I think it should?

Reply:
Some exotic and counter(first time I know about it) type related tests are added in `VerifierTests`. `Coalesce` does not reject counter type, as far as the data types of all its arguments are consistent, it is happy. If we keep it this way `where coalesce(k8s.pod.network.tx) IS NULL` does succeed without error(?).

Reply:
That's fair, but there is another aspect here that I need to point out.

Since `coalesce` has been initially implemented and until now, `counter` field types have been introduced in a restricted support kind of way: they can be displayed, kept, dropped but they cannot be used in functions and commands, unless they are casted to their respective supported numeric data type. See also https://github.com/elastic/elasticsearch/issues/111976

`coalesce`, like `is null`/`is not null`, `case`, `mv_expand` (and maybe others) accept `counter`s and this might be ok, since they do not act on the value of that `counter` field. Wdyt @nik9000 @dnhatn ?